File: js/src/trees/lowestCommonAncestor.js
class TreeNode {
	constructor(val = null, left = 0, right = 0) {
		this.val = val;
		this.left = left;
		this.right = right;
	}
}

const lowestCommonAncestor = (root, p, q) => {
	let cur = root;
	while (cur) {
		if (p.val < cur.val && q.val < cur.val) {
			cur = root.left;
		} else if (p.val > cur.val && q.val > cur.val) {
			cur = root.right;
		} else return cur;
	}
	return null;
};

File: python/trees/lowestCommonAncestor.py
from typing import Optional
from python.trees.isSame import TreeNode


class Solution:
    def lowestCommonAncestor(
        self, root: Optional[TreeNode], p: TreeNode, q: TreeNode
    ) -> Optional[TreeNode]:
        if not root:
            return None
        if p.val > root.val and q.val > root.val:
            return self.lowestCommonAncestor(root.right, p, q)
        elif p.val < root.val and q.val < root.val:
            return self.lowestCommonAncestor(root.left, p, q)
        return root
